### ISSUE TITLE
color-contrast: False positive for in-gamut OKLCH colors

### ISSUE THREAD

**Issue by rzschoch** [dequelabs/axe-core#5036]
### Product

axe-core

### Product Version

4.11.1

### Latest Version

- [x] I have tested the issue with the latest version of the product

### Expectation

When using in-gamut OKLCH colors via CSS `light-dark()` with `color-scheme: light`, I expect
axe-core's `color-contrast` rule to compute the correct sRGB equivalents and report accurate
contrast ratios matching the browser's actual rendering.

### Actual

axe-core computes significantly incorrect sRGB values for OKLCH colors, leading to false
positive contrast failures. The deviation is not a rounding error but a substantial difference:

| Token      | OKLCH                         | Browser renders         | axe-core computes       | RGB difference         |
| ---------- | ----------------------------- | ----------------------- | ----------------------- | ---------------------- |
| primary    | `oklch(47.555% 0.085 224deg)` | `#0f6680` (15,102,128)  | `#347c92` (52,124,146)  | R: +37, G: +22, B: +18 |
| surface    | `oklch(98.24% 0.001 286deg)`  | `#f9f9fa` (249,249,250) | `#eeeeef` (238,238,239) | R: -11, G: -11, B: -11 |
| on-primary | `oklch(100% 0 0deg)`          | `#ffffff` (255,255,255) | `#dde9ec` (221,233,236) | R: -34, G: -22, B: -19 |

This causes false positive contrast violations:

| Element                                    | axe-core contrast | Actual contrast | WCAG AA |
| ------------------------------------------ | ----------------- | --------------- | ------- |
| Outlined button (primary text on surface)  | 4.07:1 (fail)     | 6.17:1 (pass)   | 4.5:1   |
| Filled button (on-primary text on primary) | 4.33:1 (fail)     | 6.50:1 (pass)   | 4.5:1   |

Browser pixel colors were verified via canvas rendering:

```javascript
const canvas = document.createElement("canvas");
const ctx = canvas.getContext("2d", { colorSpace: "srgb" });
ctx.fillStyle = "oklch(47.555% 0.085 224deg)";
ctx.fillRect(0, 0, 10, 10);
const pixel = ctx.getImageData(0, 0, 1, 1).data;
// Result: [15, 102, 128, 255] → #0f6680
```

### How to Reproduce

```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<style>
			html {
				color-scheme: light;
			}
			.container {
				background: oklch(98.24% 0.001 286deg);
				padding: 2rem;
			}
			.button-outlined {
				color: oklch(47.555% 0.085 224deg);
				background: transparent;
				border: 1px solid currentColor;
				padding: 0.5rem 1rem;
				font-size: 14px;
				font-weight: bold;
			}
			.button-filled {
				color: oklch(100% 0 0deg);
				background: oklch(47.555% 0.085 224deg);
				border: none;
				padding: 0.5rem 1rem;
				font-size: 14px;
				font-weight: bold;
			}
		</style>
	</head>
	<body>
		<div class="container">
			<button class="button-outlined">Cancel</button>
			<button class="button-filled">Confirm</button>
		</div>
	</body>
</html>
```

Run axe-core on this page. It reports `color-contrast` violations for both buttons, even though
the actual browser contrasts are 6.17:1 and 6.50:1 (well above the WCAG AA minimum of 4.5:1).

### Additional context

- All OKLCH values used here are within the sRGB gamut, so no gamut mapping should be needed.
- The issue may be related to the `light-dark()` function resolution combined with OKLCH parsing.
- Tested in Chromium-based browsers (Chrome 134, Edge via Playwright).
- Related: #4908 ("Match browser behavior for out-of-gamut oklch colors", fixed in 4.11.1 but
  problem persists for in-gamut colors) and #4007 ("Axe fails with new color spec").
